addon/ExperimentProvider.js

```diff
@@ -1,60 +1,101 @@
-const seedrandom = require("common/vendor")("seedrandom");
-const simplePrefs = require("sdk/simple-prefs");
-const OVERRIDE_PREF = "experimentOverrides";
+const prefService = require("sdk/preferences/service");
+const {PrefsTarget} = require("sdk/preferences/event-target");
+const ss = require("sdk/simple-storage");
 
-exports.OVERRIDE_PREF = OVERRIDE_PREF;
 exports.ExperimentProvider = class ExperimentProvider {
-  constructor(clientID, experiments = require("../experiments.json"), rng) {
-    this._clientID = clientID;
+  constructor(experiments = require("../experiments.json"), rng) {
     this._experiments = experiments;
-    this._rng = rng || seedrandom(clientID);
+    this._rng = rng || Math.random;
     this._data = {};
     this._experimentId = null;
+    ss.storage.overrideExperimentProvider = false;
+    this._target = PrefsTarget();
   }
 
   init() {
     this.setValues();
-    this._onPrefChange = () => this.setValues();
-    simplePrefs.on(OVERRIDE_PREF, this._onPrefChange);
+    this._onPrefChange = prefName => this.overrideExperimentPrefs(prefName);
+
+    Object.keys(this._experiments).forEach(experimentName => {
+      this._target.on(experimentName, this._onPrefChange);
```

Comment:
Could we add addon namespacing to this, like simple prefs does? Setting the experiment name as-is could potentially overwrite stuff, no?